src/js/operations/Extract.js

```diff
@@ -293,5 +293,66 @@ var Extract = {
         output += Extract.run_dates(input, []);
         return output;
     },
-    
+
+    /**
+     * @constant
+     * @default
+     */
+    XPATH_INITIAL: "",
+
+    /**
+     * @constant
+     * @default
+     */
+    XPATH_DELIMITER: "\\n",
+
+    /**
+     * Extract information (from an xml document) with an XPath query
+     *
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run_xpath:function(input, args) {
+        var query = args[0];
+        var delimiter = args[1];
+
+        try {
+            var xml = $.parseXML(input);
+        } catch (err) {
+            return "Invalid input XML.";
+        }
+
+        try {
+            var result = $.xpath(xml, query);
+        } catch (err) {
+            return "Invalid XPath. Details:\n" + err.message;
+        }
+
+        var serializer = new XMLSerializer();
+        var output = "";
+        for (var i = 0; i < result.length; i++) {
```

Comment:
Hmm unfortunately `xpath()` doesn't return an array but a simple Object.
Now I have to to it this way:
~~~
        return Object.values(result).slice(0, -1) // all values except last (length)
            .map(nodeToString)
            .join(delimiter);
~~~
The `slice()` is needed to remove the length property from our map call. 
That doesn't look very clean but I can't think of a better way (except you have an idea?)